### PR TITLE
Add build icon option to `charites build`

### LINKED ISSUE #13: Add a command that can generate sprites
It is desirable to be able to maintain both icons and styles in a single project.

### PR DESCRIPTION
**Pull request by naogify**
Close #13 

Add an option for building icon's sprite image and json.

~~You can build sprite image and json by adding `--sprite` option like below.~~

```
$ charites build style.yml style.json --sprite=https://example.github.io/icons
```

~~This option will replace style.json `sprite` url and convert icon's svg files to sprites image and json file.~~

~~You can also specify the svg files directory and dist directory like below.~~

```
$ charites build style.yml --sprite <sprite url> [<icon source>] [<icon destination>]
```

~~The default `<icon source>`  is `icons/`, and the default `<icons destination>` is the directory this command runs.~~

### CODE REVIEW

**Comment by naogify** [comment]
@miya0001 

I updated the option for build map icons. Please re-review it.

CLI manual is below
```
$ charites build style.yml --sprite-url [<sprite url>] --sprite-input [<icon source>] --sprite-output [<icon destination>]
```

## Sample1: use default

```
$ charites build style.yml
```

This command will use  `./icons/*.svg` to generate files below.  Because `--sprite-input` default value is `./icons/` and `--sprite-output` default value is `./`.

Output
- `./basic.json`
- `./basic.png`
- `./basic@2x.json`
- `./basic@2x.png`
- `./style.json`

Icon name is defined by sprite url in `style.yml` . (eg, `basic.json/png` is created by sprite url `http://localhost:8888/basic`))


## Sample2 use options

```
$ charites build style.yml --sprite-url http://localhost:8888/basic-white --sprite-input ./src/icons--sprite-output ./public
```

This command will use  `./src/icons/*.svg` to generate files below.

Output
- `./public/basic-white.json`
- `./public/basic-white.png`
- `./public/basic-white@2x.json`
- `./public/basic-white@2x.png`
- `./style.json`

**Comment by miya0001** [comment]
@JinIgarashi Please review 😊

File: README.md [hunk 1]
```diff
@@ -99,6 +99,12 @@ Add `-c` or `--compact-output` to minify the JSON
 charites build style.yml style.json -c
 ```
 
+Add `--sprite-input` and `--sprite-output` to build svg files for map icons.
+
+```
+charites build style.yml --sprite-input icons/ --sprite-output public/
```
Comment (JinIgarashi) [inline README.md]: The example of command usage is wrong.

it should be like below

```diff
- charites build style.yml --sprite-input icons/ --sprite-output public/
+ charites build style.yml style.json --sprite-input icons/ --sprite-output public/
```

File: src/lib/build-sprite.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      spritezero.generateImage(option,(err:any ,result:object) => {
+          if (err) reject(err);
+          else resolve(result);
+      });
+  });
+}
+
+export async function buildSprite(svgPath: string, publicPath: string, iconSlug: string) {
+
+  const pxRatios = [1, 2]
```
Comment (JinIgarashi) [inline src/lib/build-sprite.ts:26]: `sprite@4x.json` and `sprite@4xpng` are also required to generate.

**Review by JinIgarashi — CHANGES_REQUESTED** [review]
@naogify Some comments were put, please confirm them. It works well and good job to implement including sprite generation into `build` command.

However, I don't think we need to include this implementation on `charites build` because there is no clear relationship between inputted `style.yaml` and `sprite` icons.

I suggest to create new command like `charites sprite` rather than adding existing `build` command in order to simplify the command.

Maybe, the command of generating sprite can be as below.

```
charites sprite ./icons ./sprites
```

When `--sprite-url` option is specified in `build` command, charites will check icon name on style.yml and it will warn if some icons are not included in either sprite.json or style.yml.
```
charites build style.yml style.json --sprite-url http://localhost:8080/sprite
```

This is my humble suggestion for command interface design.

**Comment by naogify** [comment]
@JinIgarashi 

Thank for your feedback! 

> However, I don't think we need to include this implementation on charites build because there is no clear relationship between inputted style.yaml and sprite icons.

The reason I used option is that I also want to add building sprite funtion to `charites serve`,  like `charites serve --sprite-url http://localhost:8888/sprite`. So, I thougt it's better to use option to create sprite file to build command.

However, I thought it would be a good idea to provide both the `charites sprite` command and the `charites serve --sprite-url http://localhost:800/sprite` option, so that user can see the changes to the SVG icon on the map in real time, and also it would also make it easier to understand how to build sprites. What do you think?

File: src/lib/build-sprite.ts (hunk 2 above)
Comment (naogify) [reply src/lib/build-sprite.ts:26]: @JinIgarashi Thank you! I will add the function to generate `sprite@4x.json` and `sprite@4xpng`.

**Review by JinIgarashi — APPROVED** [review]

**Pull request merged**